pkg/core/fs.go

```diff
@@ -71,27 +80,50 @@ func NewReadOnlyFS(bundle *Bundle, l *zap.Logger) (*ReadOnlyFS, error) {
 	// Extract the meta information needed.
 	err := Publish(context.Background(), fs.bundle)
 	if err != nil {
-		l.Error("Failed to publish bundle", zap.String("id", bundle.BundleID), zap.Error(err))
+		fs.l.Error("Failed to publish bundle", zap.String("id", bundle.BundleID), zap.Error(err))
 		return nil, err
 	}
-	// TODO: Introduce streaming and caching
-	// Populate the filesystem.
+	// Populate the filesystem with medatata
+	//
+	// NOTE: this fetches the bundle metadata only and builds
+	// an in-memory view of the file system structure.
+	// This may lead to a large memory footprint for bundles
+	// with many files (e.g. thousands)
+	//
+	// TODO: reduce memory footprint
 	return fs.populateFS(bundle)
 }
 
+func localPath(consumable fmt.Stringer) (string, error) {
```

Comment:
can we land this incremental improvement with a `TODO` or `NOTE` or `???` s'th to indicate that the methodology of extracting this info via 
* string operations on the return values from the `storage` api
rather than
* functions or methods defined in `storage`
remains a questionable design decision?

i think i suggested doing this at one point in order to make a one-idea-per-commit change, and i suppose the current implementation is better than the previous, so i'm not looking to see `storage` change here, just to get the `grep`-able tag in the comments so we recall that this could be further refined.

Reply:
ok agree. Added the TODO (the `???`  is more your distinctive mark :) ) Defered.